lib/charms/operator_libs_linux/v0/systemd.py

```diff
@@ -163,9 +188,19 @@ def service_pause(service_name: str) -> bool:
     Args:
         service_name: the name of the service to pause
     """
-    _systemctl("disable", service_name, now=True)
-    _systemctl("mask", service_name)
-    return not service_running(service_name)
+    try:
+        _systemctl("disable", service_name, now=True)
+    except SystemdError:
+        pass
```

Comment:
What's the rationale for this pass? (and the next for that matter?)

Reply:
This is in keeping with the previous behavior, which was to fail silently if the mask or disable operation failed (possibly because the lib was already masked or disabled), but check to make sure that the service was stopped.

There are definitely some cases where you wind up with a stopped service that has not actually been masked, which is not great. I'll poke at systemd a bit and see if there's a more complete way of doing things ...

Reply:
Cool! The underlying methods are idempotent. You can repeatedly mask or unmask, or enable or disable, something without systemctl throwing an error. I got rid of the tries in code.